RUFAS/biophysical/animal/ration/ration_manager.py

```
@@ -80,6 +83,60 @@ def set_ration_feeds(cls, ration_config: dict[str, Any]) -> None:
+
+        for name, ration in (
+            ("starter", cls.feedlot_starter_ration),
+            ("transition", cls.feedlot_transition_ration),
+            ("finisher", cls.feedlot_finisher_ration),
+        ):
+            if ration:
+                total_pct = sum(ration.values())
+                if abs(total_pct - 100.0) > 1e-2:
+                    raise ValueError(
+                        f"Feedlot {name} ration percentages must sum to 100.0%, got {total_pct}%"
+                    )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)